### ISSUE TITLE
Could not parse last message {"error":"Internal Server Error"}

### ISSUE THREAD

**Issue by KJ-Waller** [huggingface/chat-ui#305]
I'm trying to run the chat-ui using [text-inference-generation](https://github.com/huggingface/text-generation-inference). I've tested my inference server using the [starchat-playground](https://huggingface.co/spaces/HuggingFaceH4/starchat-playground) chat-ui, running on the same server as inference API, and there it works fairly well. The inference URL is `http://127.0.0.1:8090`.

When using chat-ui, I can get the UI itself to work, the MongoDB server is running locally as per the readme instructions, and it stores conversations, however, none of the responses come in. I also don't see any logs on the inference server when using the HF chat-ui, so I suspect something is going wrong even when sending the request. 

The chat-ui server outputs the following error:
```Could not parse last message {"error":"Internal Server Error"}
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at parseGeneratedText (/src/routes/conversation/[id]/+server.ts:181:32)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async saveMessage (/src/routes/conversation/[id]/+server.ts:95:26)
```

While the web console outputs the following error:
```
Assertion failed: Input argument is not an HTMLInputElement     onloadwff.js:71
    getFormProfile @ onloadwff.js:71
    setFieldValue @ onloadwff.js:71
    formKeydownListener @ onloadwff.js:71
Failed to load resource: the server responded with a status of 500 (Internal Server Error)    648f1b040482530a0eb4d2e7:1
Error: Internal Server Error
    at streamingRequest (index.mjs:210:15)
    at async textGenerationStream (index.mjs:467:3)
    at async getTextGenerationStream (+page.svelte:64:20)
    at async writeMessage (+page.svelt:190:27)
  writeMessage @ +page.svelte:215
```

Basically, what I think is happening is the response seems to be `{"error":"Internal Server Error"}` and the error occurs [when trying to parse `data:` from the response](https://github.com/huggingface/chat-ui/blob/5d07536967d0da65d16238dbf9e13871d128c808/src/routes/conversation/%5Bid%5D/%2Bserver.ts#L179), which doesn't exist. 

Here is what my .env.local looks like where I define the model: 
```
MODELS=`[
  {
    "name": "Self hosted Bloom",
    "description": "A good alternative to ChatGPT",
    "websiteUrl": "http://127.0.0.1:8090/generate_stream",
    "userMessageToken": "<|prompter|>",
    "assistantMessageToken": "<|assistant|>",
    "messageEndToken": "</s>",
    "preprompt": "You are an AI assistant trying to help explain things. Following is a question:\n",
    "promptExamples": [
      {
        "title": "Explain what deep learning is",
        "prompt": "Explain what deep learning is"
      }
    ],
    "parameters": {
      "temperature": 0.9,
      "top_p": 0.95,
      "repetition_penalty": 4.0,
      "top_k": 50,
      "truncate": 1000,
      "max_new_tokens": 1024
    }
  }
]`
```

**Comment by KJ-Waller** [comment]
I printed the url it's sending the request to, and it was set to `https://api-inference.huggingface.co/models/Self hosted Bloom`

I'm not sure now how I'm supposed to format the `.env.local` file since I've tried setting multiple fields such as `url` and `endpoint`, but it still prints out the huggingface api-inference url + the model name. I've also tried formatting it like so: ``` MODELS=`[{"name": "mymodel", "endpoints": [{"url": "http://127.0.0.1:8090/generate_stream"}]}]` ``` as other people in past issues have formatted custom endpoints but it still doesn't work.

**Comment by KJ-Waller** [comment]
Somehow it fixed itself. After digging through some of the JavaScript code I found [where the endpoint is determined](https://github.com/huggingface/chat-ui/blob/e34af3608b23bda963bd4f93a063fb03f20be81b/src/lib/server/modelEndpoint.ts#L13) and figured out how to format my endpoint:

```
MODELS=`[
  {
    ...
    "endpoints": [{"url": "http://127.0.0.1:8090/generate_stream"}],
    ...
  }
]`
```

I think I must have made a typo previously when trying this particular format. For anyone in the future with this issue, make sure to set `endpoints` as how I've formatted it above, and completely restart the server, not just refreshing it, as it does not seem to pick up on changes in .env.local file otherwise.
